File: kotlin-java-codingtest/src/main/kotlin/programmers/lv0/그림_확대/그림 확대.kt
package programmers.lv0.그림_확대

fun solution(picture: Array<String>, k: Int): Array<String> {
    val answer = Array(picture.size * k) { Array(picture[0].length * k) { ' ' } }

    for (i in picture.indices) {
        for (j in picture[i].indices) {
            for (x in 0 until k) {
                for (y in 0 until k) {
                    answer[i * k + x][j * k + y] = picture[i][j]
                }
            }
        }
    }

    return answer.map { it.joinToString("") }.toTypedArray()
}

File: Python/programmers/lv0/그림 확대.py
def solution(picture, k):
    answer = [i for i in range(1, len(picture) * k)]

    for i in range(0, len(picture), k):
        for j in range(0, len(picture[0]), k):
            for x in range(k):
                for y in range(k):
                    answer[i * k + j + x * k + y] = picture[i + x][j + y]

    return answer
